Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -0,0 +1,160 @@
+#:  * `extract` [`--force`] <formula> <tap> [`--version=`<version>]:
+#:    Looks through repository history to find the <version> of <formula> and
+#:    creates a copy in <tap>/Formula/<formula>@<version>.rb. If the tap is
+#:    not installed yet, attempts to install/clone the tap before continuing.
+#:
+#:    If `--force` is passed, the file at the destination will be overwritten
+#:    if it already exists. Otherwise, existing files will be preserved.
+#:
+#:    If an argument is passed through `--version`, <version> of <formula>
+#:    will be extracted and placed in the destination tap. Otherwise, the most
+#:    recent version that can be found will be used.
+
+require "cli_parser"
+require "utils/git"
+require "formulary"
+require "tap"
+
+def with_monkey_patch
+  BottleSpecification.class_eval do
+    if method_defined?(:method_missing)
+      alias_method :old_method_missing, :method_missing
+    end
+    define_method(:method_missing) { |*| }
+  end
+
+  Module.class_eval do
+    if method_defined?(:method_missing)
+      alias_method :old_method_missing, :method_missing
+    end
+    define_method(:method_missing) { |*| }
+  end
+
+  Resource.class_eval do
+    if method_defined?(:method_missing)
+      alias_method :old_method_missing, :method_missing
+    end
+    define_method(:method_missing) { |*| }
+  end
+
+  yield
+ensure
+  BottleSpecification.class_eval do
+    if method_defined?(:old_method_missing)
+      alias_method :method_missing, :old_method_missing
+      undef :old_method_missing
+    end
+  end
+
+  Module.class_eval do
+    if method_defined?(:old_method_missing)
+      alias_method :method_missing, :old_method_missing
+      undef :old_method_missing
+    end
+  end
+
+  Resource.class_eval do
+    if method_defined?(:old_method_missing)
+      alias_method :method_missing, :old_method_missing
+      undef :old_method_missing
+    end
+  end
+end
+
+class DependencyCollector
+  def parse_symbol_spec(*); end
+
+  module Compat
+    def parse_string_spec(spec, tags)
```

Comment:
What is this doing that `def parse_symbol_spec(*); end` doesn't?

Reply:
I was trying to monkey-patch over [this `compat` definition of `parse_symbol_spec](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/compat/dependency_collector.rb); the [`super` one](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dependency_collector.rb) works fine as far as I can tell.

Of course, we could just clobber over it with a no-op and I think we'd be able to move on just fine.